t/e2e/transactions/transactions_test.go

```diff
@@ -139,8 +141,51 @@ func (s *TransactionsTestSuite) TestCallRPCSendTransactionUpstream() {
 	s.Equal(`{"jsonrpc":"2.0","id":1,"result":"`+txHash.String()+`"}`, result)
 }
 
-// FIXME(tiabc): Sometimes it fails due to "no suitable peers found".
+// TestSendContractCompat tries to send transaction using the legacy "Data"
+// field, which is supported for backward compatibility reasons.
+func (s *TransactionsTestSuite) TestSendContractTxCompat() {
+	initFunc := func(byteCode []byte, args *common.SendTxArgs) {
+		args.Data = (hexutil.Bytes)(byteCode)
+	}
+	s.testSendContractTx(initFunc, nil, "")
+}
+
+// TestSendContractCompat tries to send transaction using both the legacy
+// "Data" and "Input" fields. Also makes sure that the error is returned if
+// they have different values.
+func (s *TransactionsTestSuite) TestSendContractTxCollision() {
+	// Scenario 1: Both fields are filled and have the same value, expect success
+	initFunc := func(byteCode []byte, args *common.SendTxArgs) {
+		args.Input = (hexutil.Bytes)(byteCode)
+		args.Data = (hexutil.Bytes)(byteCode)
+	}
+	s.testSendContractTx(initFunc, nil, "")
+
+	// Scenario 2: Both fields are filled with different values, expect an error
+	inverted := func(source []byte) []byte {
+		inverse := make([]byte, len(source))
+		copy(inverse, source)
+		for i, b := range inverse {
+			inverse[i] = b ^ 0xFF
+		}
+		return inverse
+	}
+
+	initFunc2 := func(byteCode []byte, args *common.SendTxArgs) {
+		args.Input = (hexutil.Bytes)(byteCode)
+		args.Data = (hexutil.Bytes)(inverted(byteCode))
```

Comment:
Not sure we need such a "complex" logic here. Maybe this func instead https://golang.org/pkg/math/rand/#Read?

Reply:
yeah, seems to be an overkill to prove that data is not equal to input

Reply:
hmm, I don't see what will we gain from there... replace 8 LoC with 6? on a positive side, this one is 100% deterministic.

Reply:
you can change first by to anything different

Reply:
do you mean inverting one byte instead of inverting an array of bytes?